lib/resource/project.js

```diff
@@ -59,66 +67,130 @@ var util = require('../common/util.js');
+     * @param {object=} config - See {module:resource#createProject}.
+     *
+     * @example
+     * project.create(function(err, zone, apiResponse) {
+     *   if (!err) {
+     *     // The zone was created successfully.
+     *   }
+     * });
+     */
+    create: true,
+
+    /**
+     * Delete the project.
+     *
+     * @resource [projects: delete API Documentation]{@link https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/delete}
+     *
+     * @private
+     *
+     * @param {function=} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * project.delete(function(err, apiResponse) {
+     *   if (!err) {
+     *     // The project was deleted!
+     *   }
+     * });
+     */
+    delete: true,
+
+    /**
+     * Check if the project exists.
+     *
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {boolean} callback.exists - Whether the project exists or not.
+     *
+     * @example
+     * project.exists(function(err, exists) {});
+     */
+    exists: true,
+
+    /**
+     * Get a project if it exists.
+     *
+     * You may optionally use this to "get or create" an object by providing an
+     * object with `autoCreate` set to `true`. Any extra configuration that is
+     * normally required for the `create` method must be contained within this
+     * object as well.
+     *
+     * @param {options=} options - Configuration object.
+     * @param {boolean} options.autoCreate - Automatically create the object if
+     *     it does not exist. Default: `false`
+     *
+     * @example
+     * project.get(function(err, project, apiResponse) {
+     *   // `project.metadata` has been populated.
+     * });
+     */
+    get: true,
+
+    /**
+     * Get the metadata for the project.
+     *
+     * @resource [projects: get API Documentation]{@link https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/get}
+     *
+     * @param {function=} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {?object} callback.metadata - Metadata of the project from the
+     *     API.
+     * @param {object} callback.apiResponse - Raw API response.
+     *
+     * @example
+     * project.getMetadata(function(err, metadata, apiResponse) {});
+     */
+    getMetadata: true,
+
+    /**
+     * Set the project's metadata.
+     *
+     * @resource [projects: update API Documentation]{@link https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/update}
+     * @resource [Project Resource]{@link https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects#Project}
+     *
+     * @private
```

Comment:
Actually, I think it's because I originally thought that the API was busted, but it's just that you have to be authenticated as yourself (not a service account). I need to just add a line about that and remove the private tag.